tooltester/src/main/java/io/dockstore/tooltester/client/cli/WorkflowRunner.java

```diff
@@ -0,0 +1,146 @@
+package io.dockstore.tooltester.client.cli;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.dockstore.common.Utilities;
+import io.dockstore.openapi.client.api.Ga4Ghv20Api;
+import io.dockstore.openapi.client.model.FileWrapper;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static com.amazonaws.util.DateUtils.parseISO8601Date;
+import static io.dockstore.tooltester.client.cli.JCommanderUtility.out;
+import static io.dockstore.tooltester.helper.ExceptionHandler.COMMAND_ERROR;
+import static io.dockstore.tooltester.helper.ExceptionHandler.errorMessage;
+import static io.dockstore.tooltester.helper.ExceptionHandler.exceptionMessage;
+import static java.util.UUID.randomUUID;
+
+
+public class WorkflowRunner {
+
+    public static final int MILLISECONDS_IN_SECOND = 1000;
+    public static final int SECONDS_IN_MINUTE = 60;
+    private String entry;
+    private String version;
+    private String runID = null;
+    private String pathOfTestParameter;
+    private String agcWrapperPath = "agcWrapper.json";
+    private JsonObject log = null;
+    private Boolean finished = false;
+    private String state;
+
+    private final List<String> inProgressStates = Arrays.asList("RUNNING", "INITIALIZING");
+
+    public WorkflowRunner(String entry, String version, String pathOfTestParameter) {
+        this.entry = entry;
+        this.version = version;
+        this.pathOfTestParameter = pathOfTestParameter;
+    }
+
+    public WorkflowRunner(String entry, String version, String relativePathToTestParameterFile, Ga4Ghv20Api ga4Ghv20Api) throws InterruptedException {
+        this.entry = entry;
+        this.version = version;
+
+        File testParameterFile = new File("test-paramter-file-" + randomUUID() + ".json");
```

Comment:
Was going to use the function `Java.io.File.createTempFile()`, but it would only give an absolute file path, and I couldn't find an elegant way to turn it into a relative file path. This is required as the `WES` command in the `CLI` only supports relative paths.

I think this solution works quite well.

Reply:
This is fine! But FYI, for future reference, there is a `java.nio.io.Files.createTempFile()` where you can specify where to create the file.

Reply:
Okay, thanks. I will do that next time.